### PR TITLE
Continue calculating dependencies when failing to obtain remote for r…

### PR DESCRIPTION
**Pull request by riarenas**
…epo due to Github App not being installed.

Also changed the exception type thrown, as it didn't seem like a SubscriptionException in this case.

### CODE REVIEW

**Comment by riarenas** [comment]
https://github.com/dotnet/arcade/issues/4694

File: src/Maestro/SubscriptionActorService/DarcRemoteFactory.cs [hunk 1]
```diff
@@ -83,7 +83,7 @@ public async Task<IRemote> GetRemoteAsync(string repoUrl, ILogger logger)
                     case "github.com":
                         if (installationId == default)
                         {
-                            throw new SubscriptionException($"No installation is avaliable for repository '{normalizedUrl}'");
+                            throw new DependencyException($"No installation is avaliable for repository '{normalizedUrl}'");
```
Comment (mmitche) [inline src/Maestro/SubscriptionActorService/DarcRemoteFactory.cs]: I think we should add a new exception type for this, as it probably could be fatal in some other cases. There may be places we catch DependencyException today that we would not want to catch in this case.

Comment (riarenas) [reply src/Maestro/SubscriptionActorService/DarcRemoteFactory.cs]: I checked and that's not the case, but it makes sense. I'll make it a new exception type.

Comment (riarenas) [reply src/Maestro/SubscriptionActorService/DarcRemoteFactory.cs]: Added a new type for issues with the Github App installation for future-proofing.

**Pull request merged**